sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -180,4 +184,60 @@ export class ApplicationController extends BaseController {
       );
     }
   }
+
+  /**
+   * Generic method to update all Student Application status from frontend.
+   * @param applicationId application id to be updated.
+   * @body payload contains the status, that need to be updated
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/update-status")
+  async updateStudentApplicationStatus(
+    @UserToken() userToken: IUserToken,
+    @Param("applicationId") applicationId: number,
+    @Body() payload: ApplicationStatusToBeUpdatedDto,
+  ): Promise<void> {
+    const student = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+    if (!student) {
```

Comment:
I understanding that this check is more specific, but IMHO the check already in place on "getApplicationByIdAndUserName" would be enough to ensure the authorization in this case, since we ensure that the application belongs to the user accessing it.

Reply:
ya, for `getApplicationByIdAndUserName` the check is not need.  for the previous method, I was using the `student`. update it.